integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CommonUtil.scala

```diff
@@ -226,11 +226,13 @@ object CommonUtil {
         value.matches(pattern)
       case "timestamptype" =>
         val timeStampFormat = new SimpleDateFormat(CarbonProperties.getInstance()
-          .getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT))
+          .getProperty(CarbonCommonConstants.CARBON_TIMESTAMP_FORMAT,
```

Comment:
Partition input should be validated wrt to CARBON_TIMESTAMP_FORMAT. It should always take SPARK_TIMESTAMP_FORMAT. CARBON_TIMESTAMP_FORMAT is applicable only for dataload csv input only.
Same for data format.